### PR TITLE
make sure stream_slices is an iterator

### PR DESCRIPTION
**Pull request by brianjlai**
## What
The `stream_slices()` interface is defined in the CDK to return `Iterable[Optional[Mapping[str, Any]]]`. This encompasses the set of all iterable types including iterators, lists, and tuples. However, in `check_connection()` we rely on the `next()` function which only works for iterators. Lists and tuples are iterables that are not iterators which will cause an iterator error:
```
{"type": "CONNECTION_STATUS", "connectionStatus": {"status": "FAILED", "message": "\"Unable to connect to stream rates - 'list' object is not an iterator\""}}
```

## How
When processing stream_slice output, we should always translate this iterable into an iterator using `iter()`

## Recommended reading order
1. `check_stream.py`

### CODE REVIEW

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/checks/check_stream.py [hunk 1]
```diff
@@ -47,9 +47,13 @@ def check_connection(self, source: Source, logger: logging.Logger, config: Mappi
         return True, None
 
     def _get_stream_slice(self, stream):
-        slices = stream.stream_slices(
-            cursor_field=stream.cursor_field,
-            sync_mode=SyncMode.full_refresh,
+        # We wrap the return output of stream_slices() because some implementations return types that are iterable,
+        # but not iterators such as lists or tuples
+        slices = iter(
```
Comment (brianjlai) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/checks/check_stream.py:52]: given that some output of `stream_slices()` could return something that is already an iterator, but calling `iter()` on an object that is already an iterator ends up with the same result. I experimented with this and added a test case for this.

The alternative is we use isinstance to check for the iterator type, but it seems like unnecessary clutter to the code that would yield the same result

Comment (girarda) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/checks/check_stream.py:52]: <3 for adding a testcase for calling `iter` on an iterator

**Review by girarda — APPROVED** [review]

**Pull request merged**